models/experimental/llama2_70b/tt/model_config.py

```diff
@@ -849,14 +826,14 @@ def get_model_config(model_config_str="BFLOAT16-DRAM", num_devices=8, seq_len=1)
             mcast_in0=True,
         )
     else:
-        model_config[
-            "SELFOUT_MM_PROGCFG_LAMBDA"
-        ] = lambda seq_tiles, cores_y: ttl.operations.primary.MatmulMultiCoreReuseMultiCastProgramConfig(
+        seq_len_tiles = seq_len // 32
+        cores_y = 4  # 8 if seq_len_tiles % 8 == 0 else 4
+        model_config["SELFOUT_MM_PROGCFG_LAMBDA"] = ttl.operations.primary.MatmulMultiCoreReuseMultiCastProgramConfig(
```

Comment:
This config is no longer a lambda. It would be good to update the name so that's clear.